robyn/openapi.py

```diff
@@ -0,0 +1,287 @@
+    @param termsOfService: Optional[str] A URL to the Terms of Service for the API.
+    @param contact: Contact The contact information for the exposed API.
+    @param license: License The license information for the exposed API.
+    @param servers: list[Server] An list of Server objects representing the servers.
+    @param externalDocs: Optional[ExternalDocumentation] Additional external documentation.
+    @param components: Components An element to hold various schemas for the document.
+    """
+
+    title: str = "Robyn API"
+    version: str = "1.0.0"
+    description: Optional[str] = None
+    termsOfService: Optional[str] = None
+    contact: Contact = field(default_factory=Contact)
+    license: License = field(default_factory=License)
+    servers: List[Server] = field(default_factory=list)
+    externalDocs: Optional[ExternalDocumentation] = field(default_factory=ExternalDocumentation)
+    components: Components = field(default_factory=Components)
+
+
+@dataclass
+class OpenAPI:
+    """
+    This is the root object of the OpenAPI document.
+
+    @param info: OpenAPIInfo Provides metadata about the API.
+    @param openapi_spec: dict The content of openapi.json as a dict
+    """
+
+    info: OpenAPIInfo = field(default_factory=OpenAPIInfo)
+    openapi_spec: dict = field(init=False)
+
+    def __post_init__(self):
+        """
+        Initializes the openapi_spec dict
+        """
+        self.openapi_spec = {
+            "openapi": "3.0.0",
+            "info": asdict(self.info),
+            "paths": {},
+            "components": asdict(self.info.components),
+            "servers": [asdict(server) for server in self.info.servers],
+            "externalDocs": asdict(self.info.externalDocs) if self.info.externalDocs.url else None,
+        }
+
+    def add_openapi_path_obj(self, route_type: str, endpoint: str, openapi_summary: str, openapi_tags: list, signature: Signature = None):
+        """
+        Adds the given path to openapi spec
+
+        :param route_type: the http method as string (get, post ...)
+        :param endpoint: the endpoint to be added
+        :param openapi_summary: short summary of the endpoint (to be fetched from the endpoint defenition by default)
+        :param openapi_tags: tags -- for grouping of endpoints
+        :param signature: the function signature -- to grab the typed dict annotations for query params
+        """
+
+        query_params = None
+
+        if signature and "query_params" in signature.parameters:
+            query_params = signature.parameters["query_params"].default
+
+        modified_endpoint, path_obj = self.get_path_obj(endpoint, openapi_summary, openapi_tags, query_params)
+
+        if modified_endpoint not in self.openapi_spec["paths"]:
+            self.openapi_spec["paths"][modified_endpoint] = {}
+        self.openapi_spec["paths"][modified_endpoint][route_type] = path_obj
+
+    def add_subrouter_paths(self, subrouter_openapi):
```

Comment:
```suggestion
    def add_subrouter_paths(self, subrouter_openapi):
```

Could you add a type here?